lib/galaxy/managers/workflows.py

```diff
@@ -818,6 +819,12 @@ def _workflow_from_raw_description(
 
         workflow.has_cycles = True
         workflow.steps = steps
+        # Safeguard: workflow was implicitly merged into this Session prior to SQLAlchemy 2.0.
```

Comment:
And I think that was a bug ? Either we add workflow or stored workflow or we don't ?
Is anything breaking if you remove this ?

Reply:
I don't know - I haven't tried running all tests without this safeguard but with the cascade disabled for this relationship. My plan is to (1) add these safeguards to mimic the current behavior, (2) verify that nothing breaks, (3) turn on the future flag on session, and (4) remove the safeguards. I'm trying to do minimal/incremental changes to simplify debugging in case things break. I'm not opposed to fixing things here, but I'd rather be too careful.